src/interpolation/ArborX_InterpMovingLeastSquares.hpp

```diff
@@ -0,0 +1,274 @@
+  }
+
+  KOKKOS_INLINE_FUNCTION static auto
+  get(Interpolation::Details::MLSTargetPointsPredicateWrapper<Points> const &tp,
+      int const i)
+  {
+    return nearest(
+        AccessTraits<Points, PrimitivesTag>::get(tp.target_points, i),
+        tp.num_neighbors);
+  }
+
+  using memory_space =
+      typename AccessTraits<Points, PrimitivesTag>::memory_space;
+};
+
+} // namespace ArborX
+
+namespace ArborX::Interpolation
+{
+
+template <typename MemorySpace, typename FloatingCalculationType>
+class MovingLeastSquares
+{
+public:
+  template <typename ExecutionSpace, typename SourcePoints,
+            typename TargetPoints, typename CRBF, typename PolynomialDegree>
+  MovingLeastSquares(ExecutionSpace const &space,
+                     SourcePoints const &source_points,
+                     TargetPoints const &target_points,
+                     std::optional<int> num_neighbors, CRBF, PolynomialDegree)
+  {
+    KokkosExt::ScopedProfileRegion guard("ArborX::MovingLeastSquares");
+
+    static_assert(
+        KokkosExt::is_accessible_from<MemorySpace, ExecutionSpace>::value,
+        "Memory space must be accessible from the execution space");
+
+    // SourcePoints is an access trait of points
+    ArborX::Details::check_valid_access_traits(PrimitivesTag{}, source_points);
+    using src_acc = AccessTraits<SourcePoints, PrimitivesTag>;
+    static_assert(KokkosExt::is_accessible_from<typename src_acc::memory_space,
+                                                ExecutionSpace>::value,
+                  "Source points must be accessible from the execution space");
+    using src_point =
+        typename ArborX::Details::AccessTraitsHelper<src_acc>::type;
+    GeometryTraits::check_valid_geometry_traits(src_point{});
+    static_assert(GeometryTraits::is_point<src_point>::value,
+                  "Source points elements must be points");
+    static constexpr int dimension = GeometryTraits::dimension_v<src_point>;
+
+    // TargetPoints is an access trait of points
+    ArborX::Details::check_valid_access_traits(PrimitivesTag{}, target_points);
+    using tgt_acc = AccessTraits<TargetPoints, PrimitivesTag>;
+    static_assert(KokkosExt::is_accessible_from<typename tgt_acc::memory_space,
+                                                ExecutionSpace>::value,
+                  "Target points must be accessible from the execution space");
+    using tgt_point =
+        typename ArborX::Details::AccessTraitsHelper<tgt_acc>::type;
+    GeometryTraits::check_valid_geometry_traits(tgt_point{});
+    static_assert(GeometryTraits::is_point<tgt_point>::value,
+                  "Target points elements must be points");
+    static_assert(dimension == GeometryTraits::dimension_v<tgt_point>,
+                  "Target and source points must have the same dimension");
+
+    int num_neighbors_val =
+        (!num_neighbors)
+            ? Details::polynomialBasisSize<dimension, PolynomialDegree::value>()
+            : *num_neighbors;
```

Comment:
```suggestion
        num_neighbors
            ? *num_neighbors
            : Details::polynomialBasisSize<dimension, PolynomialDegree::value>();
```
to improve readability some.